cypress/integration/console/gateways/general-settings.spec.js

```diff
@@ -0,0 +1,123 @@
+      .and('eq', gateway.ids.gateway_id)
+    cy.findByLabelText('Gateway EUI')
+      .should('not.be.disabled')
+      .and('have.attr', 'value')
+      .and('eq', gateway.ids.eui)
+    cy.findByLabelText('Gateway Name')
+      .should('be.visible')
+      .and('have.attr', 'value')
+      .and('eq', gateway.name)
+    cy.findByLabelText('Gateway description')
+      .should('be.visible')
+      .and('have.text', gateway.description)
+    cy.findDescriptionByLabelText('Gateway description')
+      .should(
+        'contain',
+        'Optional gateway description; can also be used to save notes about the gateway',
+      )
+      .and('be.visible')
+    cy.findByLabelText('Gateway Server address')
+      .should('be.visible')
+      .and('have.attr', 'value', gateway.gateway_server_address)
+    cy.findDescriptionByLabelText('Gateway Server address')
+      .should('contain', 'The address of the Gateway Server to connect to')
+      .and('be.visible')
+    cy.findDescriptionByLabelText('Automatic updates')
+      .should('contain', 'Gateway can be updated automatically')
+      .and('be.visible')
+    cy.findByLabelText('Channel')
+      .should('be.visible')
+      .and('have.attr', 'placeholder')
+      .and('eq', 'Stable')
+    cy.findDescriptionByLabelText('Channel')
+      .should('contain', 'Channel for gateway automatic updates')
+      .and('be.visible')
+    cy.findByText('LoRaWAN options', { selector: 'h3' }).should('be.visible')
+    cy.findByText('LoRaWAN network-layer settings', { selector: 'p' }).should('be.visible')
+    cy.findByLabelText('Frequency plan').should('not.exist')
+    cy.findByRole('button', { name: 'Expand' }).click()
+    cy.findByLabelText('Frequency plan').should('be.visible')
+    cy.findByLabelText('Duty cycle')
+      .should('be.visible')
+      .and('have.attr', 'value', 'true')
+    cy.findDescriptionByLabelText('Duty cycle')
+      .should('contain', 'Recommended for all gateways in order to respect spectrum regulations')
+      .and('be.visible')
+    cy.findByTestId('unit-input')
+      .should('be.visible')
+      .and('have.attr', 'value', '0.523')
+    cy.findByRole('button', { name: 'Save changes' }).should('be.visible')
+    cy.findByRole('button', { name: /Delete gateway/ }).should('be.visible')
+  })
+
+  it('shows warning modal on click delete button', () => {
+    cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+    cy.visit(
+      `${Cypress.config('consoleRootPath')}/gateways/${gateway.ids.gateway_id}/general-settings`,
+    )
+    cy.findByRole('button', { name: /Delete gateway/ }).click()
+    cy.findByTestId('modal-window')
+      .should('be.visible')
+      .within(() => {
+        cy.findByText('Delete gateway', { selector: 'h1' }).should('be.visible')
+        cy.findByText(
+          `Are you sure you want to delete "${gateway.name}"? This action cannot be undone and it will not be possible to reuse the gateway ID.`,
+        ).should('be.visible')
```

Comment:
Avoid checking by exact text. I don't want to update tests every time we change wording.